aws/rust-runtime/aws-runtime/src/env_config.rs

```diff
@@ -222,11 +262,19 @@ impl<'a> EnvConfigValue<'a> {
                 )
             });
 
-            let value = service_config.or(global_config);
-            tracing::trace!("ENV value = {value:?}");
-            value
+            if let Some(v) = service_config {
+                tracing::trace!("(service env) {env_var} = {v:?}");
+                Some(v)
+            } else if let Some(v) = global_config {
+                tracing::trace!("(global env) {env_var} = {v:?}");
+                Some(v)
+            } else {
+                tracing::trace!("(env) no value set for {env_var}");
+                None
+            }
         });
 
+        tracing::trace!("loading profile config from {profiles:?}");
```

Comment:
Maybe we should return early if `env_value` is `Some` at line 276? The reason for this is that a trace log may look a bit weird if this line says "loading profile..." but at the end of this method `env_value.or(profile_value)` ends up using a value from `env_value`.

Reply:
I'll remove the two "loading ..." messages and keep the log-before-return messages.